### DISCUSSION TITLE
Fermentation - setup and programming

### DISCUSSION THREAD

**Discussion by JohnDoeDK** [BernhardSchlegel/BierBot-Bricks#34] [General]
Hi there!

Bierbot is looking very interesting! I haven't started yet, cause I have a couple of questions:

Currently, I am brewing on a Grainfather, so that end of the brewery is taken care of.
But I'm looking for a fermentation setup that controls both heating and cooling. Obviously, the Sonoff way seems to be the easiest way although i came here because I have a leftover Raspberry Pi.

From what I can understant, you need 2 bricks, right? One for cooling and one for heating. Do I only need one temperature probe?

And regarding programming, does the software have a refrigeration protection built in?
Can you program fermentation steps in regard to temperature vs. Tilt/iSpindel (SG) vs. time?

**Comment by BernhardSchlegel** [comment]
Hey John,

glad you found the project :) Also interesting to hear your background and how you got hear. The Sonoff will be the easier option by a large margin since the building effort is close to 0.

Regarding your questions:

1. Yes, you'd need to Bricks.
2. Yes, you'll need only one temperature probe. The software allows you to freely assign any relays or probes to any devices freely. Just a note: If you have two locations in your fermentation vessel, you CAN install two probes. The reading will be averaged in this case.
3. Currently, you can only set a larger hysteresis to avoid the the cooling is turned on too often. But I'll add a possibility to set the "cooldowntime" explicitly. What exactly do you mean by "refrigeration proctection"?
4. The next step during fermentation can be trigger either manually, after the temperature was reached for a set time, after time has passed (without reaching the set temperature) and after the SG reached a certain value. So in short: yes.

One more note: You might have trouble finding the Sonoff TH (as it is end of production). The sucessor is the Sonoff TH Origin. The software for that is available here https://github.com/BernhardSchlegel/Brick-32.

Cheers,
Bernhard.

**Reply by JohnDoeDK** [reply]
1: Ok Thanks
2: Nice!
3: If a refrigerator starts to often, it can break the motor. Refrigerators are built in such a way that it can't turn on to often. But if you cut the power to the fridge, you can unwillingly reset that protection time. 
For instance, STC1000 (if you don't know what that is, Google it) has this protection time built in, and you can even set the amount of time.
4: Nice too!

By the way, would you be interested in me translating this software into Danish?

**Reply by BernhardSchlegel** [reply]
Hey John! Regarding #3: I think my assumption was right then and the provided answer fit. Current solution: Set the hysteresis to a larger value (say 1.5°C). And in the future you can use the explicit grace periode in minutes.

I'd be very interested. I can provide you JSONs with a machine translated Danish version. Your effort would reduce to "checking" instead of translating. If you're interested, shoot me a message: hello[äd] domain.com where "domain" is "bierbot".
